### DISCUSSION TITLE
Error while indexing a fresh opengrok instance for the first time.

### DISCUSSION THREAD

**Discussion by tchikh411** [oracle/opengrok#4395] [General]
**Describe the bug**
I setup the development environment of OpenGrok in my workstation running Linux. Initially the server configuration is not set. As per the documentation - https://github.com/oracle/opengrok/wiki/Developer-intro,
I setup the IDEA and when running the 
`java -Djava.util.logging.config.file=logging.properties \
    -cp 'distribution/target/dist/*' org.opengrok.indexer.index.Indexer \
    -W /var/opengrok/etc/configuration.xml \
    -s /var/opengrok/src \
    -d /var/opengrok/data \
    -c /usr/local/bin/ctags \
    -H -S -P \
    -U http://localhost:8080/source
`
I am getting below error:
`
SEVERE: cannot reach OpenGrok web application on http://localhost:8080/source: Internal Server Error
http://localhost:8080/source is not reachable and the -U option was specified, exiting.
`

This is because the target webappURI is not pinging the webapp as it is not setup yet. The ping is returning the 500 error code which is failing the first time index.

OpenGrok: v1.12.12, 
JDK: 11
OS: Ubuntu 22.10
Tomcat: 10.x

**To Reproduce**
Run the above index command on fresh instance

**Expected behavior**
The index should run without errors

**Comment by vladak** [comment]
The webapp will return the 500 error even when not configured, i.e. the configuration file is missing. Is there anything relevant in the Tomcat logs ?

**Comment by vladak** [comment]
It is a question whether the webapp without configuration file should return such error. Anyhow, run the indexer first to generate the config file, then deploy the webapp.

**Comment by tchikh411** [comment]
Thanks @vladak. I will check the tomcat logs and  I will give it a try.
